### PR TITLE
[GUI][Bug] Don't set a proposal's status to PASSING_NOT_FUNDED prematurely 

### PR DESCRIPTION
**Pull request by Fuzzbawls**
Simply checking that there are more "yes" votes than "no" votes isn't
enough context to determine if a proposal should be in a passing state
(`PASSING` or `PASSING_NOT_FUNDED`)

Also, Set a unique label text for proposals that are passing but there aren't
enough funds left in the budget to find them.

-----------

Also includes some very light refactoring/cleanup in related files and a compile-time warning fix.

There is some follow-up refactoring to be done, but will do that in a separate PR as it falls out-of-scope for this PR.

### CODE REVIEW

**Review by random-zebra — APPROVED** [review]
utACK 55f20c3b5446153c91f68ece080f3d5f771b1b0c

File: src/qt/pivx/governancemodel.cpp [hunk 1]
```diff
@@ -54,6 +54,7 @@ ProposalInfo GovernanceModel::buildProposalInfo(const CBudgetProposal* prop, boo
     // Calculate status
     int votesYes = prop->GetYeas();
     int votesNo = prop->GetNays();
+    int mnCount = mnodeman.CountEnabled();
```
Comment (furszy) [inline src/qt/pivx/governancemodel.cpp:57]: This loops over the masternodes maps for every existent proposal (legacy and deterministic). Locking the masternodes manager mutex on the GUI thread (which isn't good as other threads are accessing this mutex concurrently as well, which would freeze the GUI until the resource is available).

For the solution: we are already caching the masternodes count inside the `clientModel`(updating it on a timer thread), but we do it only as a string. This caches the int value: https://github.com/furszy/PIVX/commit/94aef020d5a643e1a3906654879a696b585d8714 

Feel free to squash it on your commit.

**Review by furszy — COMMENTED** [review]
Concept ACK

File: src/qt/pivx/governancemodel.h [hunk 2]
```diff
@@ -143,7 +143,7 @@ class GovernanceModel : public QObject
 
 public Q_SLOTS:
     void pollGovernanceChanged();
-    void txLoaded(const QString& hash, const int txType, const int txStatus);
+    void txLoaded(const QString& hash, int txType, int txStatus);
```
[the remark this answers is not in the document]
Comment (Fuzzbawls) [reply src/qt/pivx/governancemodel.h:146]: done

File: src/qt/pivx/governancemodel.cpp (hunk 1 above)
Comment (Fuzzbawls) [reply src/qt/pivx/governancemodel.cpp]: yeah, this was one of the follow-ups i had in mind, but i suppose it's ok to include it here directly....so, done

**Comment by Fuzzbawls** [comment]
I've extracted out the const qualifier changes as they are not directly tied to the bug that this PR fixes, and there are many more instances of such usage within our codebase that would be better handled in a dedicated PR.

File: src/qt/pivx/proposalcard.cpp [hunk 3]
```diff
@@ -59,10 +59,12 @@ void ProposalCard::setProposal(const ProposalInfo& _proposalInfo)
...
         setStatusAndVotes(tr("No Votes"), 50);
-    } else if (proposalInfo.status == ProposalInfo::NOT_PASSING ||
-        proposalInfo.status == ProposalInfo::PASSING_NOT_FUNDED) {
+    } else if (proposalInfo.status == ProposalInfo::NOT_PASSING) {
         cssClassStatus = "card-status-not-passing";
-        setStatusAndVotes(tr("Not Passing"), (int)percentageNo);
+        setStatusAndVotes( tr("Not Passing"), (int)percentageNo);
+    } else if (proposalInfo.status == ProposalInfo::PASSING_NOT_FUNDED) {
+        cssClassStatus = "card-status-not-passing";
+        setStatusAndVotes( tr("Over Budget"), (int)percentageNo);
```
Comment (furszy) [inline src/qt/pivx/proposalcard.cpp:67]: nit: extra whitespaces

**Review by furszy — DISMISSED** [review]
Sounds good, ACK 52cbb9e67e6d60b2f614d5b71b093f5a686ff06f

**Review by random-zebra — APPROVED** [review]
ACK 8a8e089d1263012cb22afe39ff565bf485705bba

**Review by furszy — APPROVED** [review]
ACK 8a8e089 and merging..

Side note, GA failure is fixed in #2722.

**Pull request merged**